lib/components/position_component.dart

```diff
@@ -25,23 +24,30 @@ import 'component.dart';
 /// They are translated by this component's (x,y). They do not need to fit
 /// within this component's (width, height).
 abstract class PositionComponent extends Component {
+  /// The position of this component on the screen (measured from the top left corner).
+  Vector2 position = Vector2.zero();
+
   /// X position of this component on the screen (measured from the top left corner).
-  double x = 0.0;
+  double get x => position.x;
+  set x(double x) => position.x = x;
 
   /// Y position of this component on the screen (measured from the top left corner).
-  double y = 0.0;
+  double get y => position.y;
+  set y(double y) => position.y = y;
 
-  /// Angle (with respect to the x-axis) this component should be rendered with.
-  /// It is rotated around its anchor.
-  double angle = 0.0;
+  /// The size that this component is rendered with.
+  /// This is not necessarily the source size of the asset.
+  Vector2 size = Vector2.zero();
 
   /// Width (size) that this component is rendered with.
-  /// This is not necessarily the source width of the asset.
-  double width = 0.0;
+  double get width => size.x;
 
   /// Height (size) that this component is rendered with.
-  /// This is not necessarily the source height of the asset.
-  double height = 0.0;
+  double get height => size.y;
```

Comment:
we can have setters for width height as well

Reply:
Good idea